src/hooks/coreSDK/useInitSafeCoreSDK.ts

```diff
@@ -23,7 +29,7 @@ export const useInitSafeCoreSDK = (): Error | null => {
         setSafeSDK(undefined)
         setError(e)
       })
-  }, [chain, wallet, safe, safeLoaded])
+  }, [chainId, wallet?.provider, wallet?.chainId, safe, safeLoaded])
 
   return error
```

Comment:
It also seems that we are not using this `error` at all thats why we didn't detect it. Might be worth logging the error to sentry or at least the console just in case.